woocommerce/woocommerce-ios
# [Composite Products] Navigate from component list to settings

Part of #8956
⚠️ Depends on #9295


## Description

This PR adds navigation from the product details Components list to the Component Settings view for composite products.

For now, the Component Settings view will show placeholders for the Component Options and Default Option; this data will need to be loaded from storage/remote (in a separate PR).

### Changes

* Adds support for initializing the `ComponentSettingsViewModel` from a `ComponentsListViewModel.Component`.
   * Includes adding the necessary properties (`description`, `optionType`) to that `Component` struct, to pass to the settings view model.
   * For now, the `options` and `defaultOptionTitle` properties in the settings VM are set to empty/placeholders.
* Adds an extension for `CompositeComponentOptionType` to display a localized description for the option type.
* Adds the navigation from `ComponentsList` to `ComponentSettings` upon selecting a component in the list.

## Testing instructions


### Prerequisites

1. Install and activate the [Composite Products extension](https://woocommerce.com/products/composite-products/) on your store.
2. Create at least one product with the Composite Products type, with at least one component.

### To test

1. Build and run the app.
2. Go to the Products tab.
3. Open a Composite Product.
4. Select the Components row in product details.
5. Select a component from the list and confirm it opens the Component Settings with the expected details (image, title, description, and placeholders for component options and default option).

## Screenshots


https://user-images.githubusercontent.com/8658164/227608095-ee090b80-f142-4ada-819a-586940ef9b4e.mp4



---
